src/Common/tests/Tests/System/PasteArgumentsTests.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System;
+
+using Xunit;
+
+namespace Tests.System
+{
+    public class PasteArgumentsTests
+    {
+        [Theory]
```

Comment:
Might want add a couple of tests for the OS-specific behavior on argv0: You can use these custom attributes

   `[PlatformSpecific(TestPlatforms.Windows)]`
   `[PlatformSpecific(TestPlatforms.AnyUnix)]`

to make the tests OS-specific.